### PR TITLE
[Merged by Bors] - Update discv5 to expand ipv6 support

### PR DESCRIPTION
**Pull request by divagant-martian**
Done in different PRs so that they can reviewed independently, as it's likely this won't be merged before I leave

Includes resolution for #4080 
- [ ] #4299
- [ ] #4318
- [ ] #4320

### CODE REVIEW

File: beacon_node/lighthouse_network/src/discovery/mod.rs [hunk 1]
```diff
@@ -428,29 +418,11 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
...
-                    .map_err(|e| format!("{:?}", e))?;
-            }
+        const IS_TCP: bool = false;
+        if self.discv5.update_local_enr_socket(socket_addr, IS_TCP) {
+            // persist modified enr to disk
+            enr::save_enr_to_disk(Path::new(&self.enr_dir), &self.local_enr(), &self.log);
         }
-
-        // replace the global version
-        *self.network_globals.local_enr.write() = self.discv5.local_enr();
```
Comment (AgeManning) [inline beacon_node/lighthouse_network/src/discovery/mod.rs:451]: Should we still update the network globals here? Is there a reason this is removed?

**Review by AgeManning — COMMENTED** [review]
This looks good to me. 

I want to just test this against some networks and do a discv5 release before we merge tho. So we have a clean discv5 target. 

There was some potentially spicy things in discv5 master at the moment that I want to check still works. 

Looking good tho!

File: beacon_node/lighthouse_network/src/discovery/mod.rs (hunk 1 above)
Comment (divagant-martian) [reply beacon_node/lighthouse_network/src/discovery/mod.rs:451]: yep you are right, ty https://github.com/sigp/lighthouse/pull/4319/commits/7a0039218b99e33aa8cd132c3be3636382c47fca

**Comment by divagant-martian** [comment]
@AgeManning checking with Pawan the doppelganger tests are failing due to an issue with geth and are not related to these changes. Asking for your review again since I changed some stuff around the boot node flags

**Review by AgeManning — APPROVED** [review]
Looks good to me. Lets do some testing, a discv5 release and then meeeeeeeerge!

**Pull request closed without merging**